include/zephyr/drivers/bluetooth.h

```diff
@@ -0,0 +1,257 @@
+ *
+ * @param dev HCI device
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_close(const struct device *dev)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->close == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->close(dev);
+}
+
+/**
+ * @brief Send HCI buffer to controller.
+ *
+ * Send an HCI packet to the controller. The packet type of the buffer
+ * must be set using bt_buf_set_type().
+ *
+ * @note This function must only be called from a cooperative thread.
+ *
+ * @param dev HCI device
+ * @param buf Buffer containing data to be sent to the controller.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_send(const struct device *dev, struct net_buf *buf)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	return api->send(dev, buf);
+}
+
+#if defined(CONFIG_BT_HCI_SETUP) || defined(__DOXYGEN__)
+/**
+ * @brief HCI vendor-specific setup
+ *
+ * Executes vendor-specific commands sequence to initialize
+ * BT Controller before BT Host executes Reset sequence. This is normally
+ * called directly after bt_hci_open().
+ *
+ * @note @kconfig{CONFIG_BT_HCI_SETUP} must be selected for this
+ * field to be available.
+ *
+ * @return 0 on success or negative error number on failure.
+ */
+static inline int bt_hci_setup(const struct device *dev, struct bt_hci_setup_params *params)
+{
+	const struct bt_hci_driver_api *api = (const struct bt_hci_driver_api *)dev->api;
+
+	if (api->setup == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->setup(dev, params);
+}
+#endif
+
+/**
+ * @}
+ */
+
+/* The following functions are not strictly part of the HCI driver API, in that
+ * they do not take as input a struct device which implements the HCI driver API.
+ */
+
+/**
+ * @brief Setup the HCI transport, which usually means to reset the
+ * Bluetooth IC.
+ *
+ * @note A weak version of this function is included in the H4 driver, so
+ *       defining it is optional per board.
+ *
+ * @param dev The device structure for the bus connecting to the IC
+ *
+ * @return 0 on success, negative error value on failure
+ */
+int bt_hci_transport_setup(const struct device *dev);
+
+/**
+ * @brief Teardown the HCI transport.
+ *
+ * @note A weak version of this function is included in the IPC driver, so
+ *		defining it is optional. NRF5340 includes support to put network core
+ *		in reset state.
+ *
+ * @param dev The device structure for the bus connecting to the IC
+ *
+ * @return 0 on success, negative error value on faulure
+ */
+int bt_hci_transport_teardown(const struct device *dev);
+
+/** Allocate an HCI event buffer.
+ *
+ * This function allocates a new buffer for an HCI event. It is given the
+ * event code and the total length of the parameters. Upon successful return
+ * the buffer is ready to have the parameters encoded into it.
+ *
+ * @param evt        Event OpCode.
```

Comment:
Maybe add a reference to the possible opcodes. I assume it's the ones defines by `hci_types.h`, right?

Reply:
Well, it's really the HCI specification that defines these, e.g. hci_types.h might be missing some.

Reply:
Ah, good point. In any case, I think referring to the define values are helpful for lookups, as we have for most other spec-defined values